RecoTauTag/RecoTau/plugins/PFRecoTauEnergyAlgorithmPlugin.cc

```diff
@@ -144,9 +154,17 @@ void PFRecoTauEnergyAlgorithmPlugin::operator()(PFTau& tau) const
 	chargedHadron != chargedHadrons.end(); ++chargedHadron ) {
     if ( chargedHadron->algoIs(PFRecoTauChargedHadron::kTrack) ) {
       ++numNonPFCandTracks;
-      const edm::Ptr<Track>& chargedHadronTrack = chargedHadron->getTrack();
-      nonPFCandTracksSumP += chargedHadronTrack->p();
-      nonPFCandTracksSumPerr2 += getTrackPerr2(*chargedHadronTrack);
+      const reco::Track* chargedHadronTrack = getTrackFromChargedHadron(*chargedHadron);
+      if ( chargedHadronTrack != nullptr ) {
+	nonPFCandTracksSumP += chargedHadronTrack->p();
+	nonPFCandTracksSumPerr2 += getTrackPerr2(*chargedHadronTrack);
+      } else {
+	edm::LogWarning("PFRecoTauEnergyAlgorithmPlugin::operator()")
+	  << "PFRecoTauChargedHadron has no associated reco::Track !!" << std::endl;
```

Comment:
endl is not needed.

If this message is "normal" for some class of candidates, then this LogWarning should be changed to LogInfo.

Reply:
There should always be a track here since the charged hadron was made from a track (check 8 lines above), both for regular RECO and MiniAOD-based RECO (lostTracks should provide a pseudoTrack()).